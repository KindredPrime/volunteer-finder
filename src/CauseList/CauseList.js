import React from 'react';
import PropTypes from 'prop-types';

function CauseList(props) {
  return (
    <div className="CauseList">
      <ul>
        {props.causes.map((cause) => (
          <li key={cause.id}>
            {cause.name}
          </li>
        ))}
      </ul>
    </div>
  );
}

CauseList.propTypes = {
  causes: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number,
    name: PropTypes.string
  })).isRequired
};

export default CauseList;