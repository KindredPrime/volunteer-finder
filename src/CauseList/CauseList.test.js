import ReactDOM from 'react-dom';
import CauseList from './CauseList';

describe('CauseList Component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <CauseList causes={[
        {
          id: 0,
          name: 'name'
        }
      ]}/>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });
});