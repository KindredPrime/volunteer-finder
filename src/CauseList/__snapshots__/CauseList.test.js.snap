// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CauseList Component renders the UI as expected 1`] = `
<body>
  <div>
    <div
      class="CauseList"
    >
      <ul>
        <li>
          Youth
        </li>
        <li>
          Health
        </li>
        <li>
          Disaster Response
        </li>
        <li>
          Animals
        </li>
        <li>
          History
        </li>
        <li>
          Education
        </li>
        <li>
          Human Rights
        </li>
        <li>
          Arts
        </li>
        <li>
          Wildlife and Environment
        </li>
        <li>
          Elderly
        </li>
        <li>
          Poverty
        </li>
      </ul>
    </div>
  </div>
</body>
`;
